File: pydss/control/ControlAction.py


class ControlAction(object):
    NONE = [1]
    OPEN = [2]
    CLOSE = [3]
    CTRL_RESET = [4]
    LOCK = [5]
    UNLOCK = [6]
    TAPUP = [7]
    TAPDOWN = [8]
    code = None

    def __init__(self, code):
        self.code = code

    def code(self):
        return self.code

    _values = [NONE, OPEN, CLOSE, CTRL_RESET, LOCK, UNLOCK, TAPUP, TAPDOWN]

    @classmethod
    def values(cls):
        return cls._enum_values[:]


ControlAction._enum_values = [ControlAction(*v) for v in ControlAction._enum_values]

File: core/jdss/src/main/java/com/ncond/dss/control/impl/ControlAction.java
package com.ncond.dss.control.impl;

public enum ControlAction {

	NONE       (1),
	OPEN       (2),
	CLOSE      (3),
	CTRL_RESET (4),
	LOCK       (5),
	UNLOCK     (6),
	TAPUP      (7),
	TAPDOWN    (8);

	private final int code;

	ControlAction(int code) {
		this.code = code;
	}

	public int code() {
		return code;
	}

}
